components/GymPlan/WorkoutSet.js

```diff
@@ -1,117 +1,68 @@
 import styled from "styled-components";
+import ExerciseSet from "./ExerciseSet";
+import { useState } from "react";
 
 export default function WorkoutSet({ setName }) {
+  const [InputFields, setInputFields] = useState(0);
+  const newSet = [];
+
+  for (let i = 1; i < InputFields; i++) {
+    newSet.push(<ExerciseSet key={i} number={i} />);
+  }
+
+  function addComponent() {
+    setInputFields((count) => count + 1);
+  }
```

Comment:
I dont really understand what's happening here. Maybe you could explain it to me some time :)

Reply:
Currently, the state & addComponent function is there for the "add exercise" button. It adds a new set for the section, where it's needed. I think I need to have a closer look on this part, to actually describe it better 😅
But I think I also need to refactor it, because it's currently just adding up as an Object Array, but doesn't differ it from each section.